test/tools/test_tools_target.py

```diff
@@ -1,10 +1,54 @@
 """Tests for targets: Channel & User"""
 from __future__ import annotations
 
+import pytest
+
 from sopel import plugin
 from sopel.tools import Identifier, target
 
 
+def test_user():
+    nick = Identifier('Brian')
+    username = 'brian'
+    host = 'example.com'
+    user = target.User(nick, username, host)
+
+    assert user.nick == nick
+    assert user.user == username
+    assert user.host == host
+    assert user.realname is None
+    assert user.channels == {}
+    assert user.account is None
+    assert user.away is None
+    assert user.hostmask == '%s!%s@%s' % (nick, username, host)
+
+
+def test_user_eq():
+    nick = Identifier('Brian')
+    username = 'brian'
+    host = 'example.com'
+    user = target.User(nick, username, host)
+
+    assert user != nick
+    assert user == target.User(nick, None, None)
+    assert user == target.User(nick, username, None)
+    assert user == target.User(nick, None, host)
+    assert user == target.User(nick, username, host)
+    assert user != target.User(Identifier('Mandy'), username, host)
+
+
+def test_user_sortable():
+    arthur = Identifier('Arthur')
+    brian = target.User(Identifier('Brian'), None, None)
+    mandy = target.User(Identifier('Mandy'), None, None)
+    reg = target.User(Identifier('Reg'), None, None)
+
+    assert sorted([brian, reg, mandy]) == [brian, mandy, reg]
+
+    with pytest.raises(TypeError):
+        sorted([brian, arthur, reg, mandy])
```

Comment:
This makes more sense to me as a direct test of comparison

```suggestion
def test_user_comparison():
    brian = target.User(Identifier('Brian'), None, None)
    mandy = target.User(Identifier('Mandy'), None, None)
    reg = target.User(Identifier('Reg'), None, None)

    assert brian < mandy < reg

    arthur = Identifier('Arthur')
    with pytest.raises(TypeError):
        arthur < brian
```